pkg/controller/auth_method.go

```diff
@@ -390,6 +390,8 @@ func waitUntilVaultPolicyBindingIsReady(c policycs.PolicyV1alpha1Interface, vpb
 
 		if vpb.Status.Phase == policyapi.PolicyBindingSuccess {
 			return true, nil
+		} else if vpb.Status.Phase == policyapi.PolicyBindingFailed {
+			return true, errors.New(fmt.Sprintf("VaultPolicyBinding %s/%s is failed", vpb.Namespace, vpb.Name))
```

Comment:
Also, return the same for VaultPolicy

Reply:
Also, do the same for VaultServer:

if !kmapi.IsConditionTrue(vs.Status.Conditions,  kmapi.ConditionFailed) {
return true, error.()
}